mlrun/runtimes/sparkjob/spark3job.py

```diff
@@ -481,6 +482,16 @@ def _get_igz_deps(self):
             "files": ["local:///igz/java/libs/v3io-pyspark.zip"],
         }
 
+    @deprecated(
+        version="1.9.0",
+        reason=(
+            "Due to Spark's restriction that node selectors "
+            "cannot be defined for both the application and driver/executor, "
+            "this method will be deprecated in 1.9.0. Use 'with_executor_node_selection()'"
```

Comment:
```suggestion
            "this method is deprecated in 1.7.0 and will be removed in 1.9.0. Use 'with_executor_node_selection()'"
```

Reply:
Why do we deprecate it instead of keeping it as a syntax sugar for setting both executor and driver NS?

Reply:
Saar suggested deprecating it, likely because it's not clear to users what this method actually does, and since in Spark, setting node selection on the function itself has no real significance. What Spark ultimately does is apply the function node selector to the driver and executor if nothing else is set. In this implementation, we're doing this ourselves to prevent users from failing Spark due to poorly defined node selectors.
Note that it also raises errors for all types other than `node_selector`, which eventually call other methods that can set this other types (affinity, tolerations, etc.) for the driver and executor, so this is a bit of a weird behavior to support it like this. I think mostly it is an unclear terminology, since users might expect this method to update the function spec directly.

However, since the method is derived from the base class, we can't simply remove it. So perhaps we could follow your suggestion and add a warning to inform users about its actual behavior. I'm not entirely sure what the best approach is here.
@alonmr, wdyt?

Reply:
Okay, talked to Liran and we agreed to have this as a syntax sugar, and pass also the other parameters to the driver and executor functions, will remove the deprecation.